CONTRIBUTING.md

```diff
@@ -29,26 +31,119 @@ The [PyScript Code of Conduct](https://github.com/pyscript/governance/blob/main/
+xs = engine.fetch_plugins(t)
+
+...
+# another 20 lines later...
+for x in xs:
+    x.init()
+```
+
+is harder to read compare to something like:
+
+```python
+plugin_type = "python"
+...
+# 20 lines later...
+plugins = engine.fetch_plugins(plugin_type)
+
+...
+# another 20 lines later...
+for plugin in plugins:
+    plugin.init()
+```
+
+**Docstrings**: Document. Your. Code! Make sure all your classes and functions have docstrings that document their behaviour, inputs and outputs. Other people (especially users!) shouldn't need to dive your the code implementing a function in order to understand what it does.
+
+Python docstrings should ideally be styled as in [PEP 8](https://peps.python.org/pep-0008/) and [PEP 257](https://peps.python.org/pep-0257/). TypeScript docstrings should be formatted per [TSDoc.org](https://tsdoc.org/). JavaString docstrings should be formatted per [JSDoc.app](https://jsdoc.app/). In general, follow the style of the existing docstrings and you'll be fine.
+
+**Comments**: In order to support the 2 previous points (remember, we said support, not replace!), make sure your core has meaningful comments that complement the code itself so that anyone reading your code can understand the intentions and the algorithm being implemented. While comments easily become out dated, they do help understanding the intentions.
+
+A PR without the above will likely be kindly rejected. :)
+
+## PR approval process
+
+**Reviewers and Approvals**: In order for a PR to be approved it has to be reviewed and approved by at least 1 maintainer, preferably 2.
+
+**Review Process**: During the PR review process, the reviewer may ask the author for changes of just have questions in general. It is the PR author responsibility to make sure they address any questions or changes requests and re-submit the PR for review. The author should not expect that PRs pending that process will be taken over by the project `maintainers`. If one or more reviewers express concern around the PR, the default process is to not merge until the concern has been addressed (there are exceptions to this rule that will be mentioned in the items below).
+
+**Meaningful Feedback**: Reviewers providing feedback (especially when expressing concerns) should make sure their feedback is backed by meaningful and clear comments and is backed by data (i.e. examples and use cases that relate to the expressed concerns). If the reviewer is in a hurry and cannot provide said feedback right away, they can quickly mention their concern and have a grace period of 1 day to amend their comment with meaningful comments and data. After the grace period is over, if the reviewer hasn't provided additional data and there are no other active concerns, the PR can be considered approved and can be merged.
+
+**Stale PRs**: PRs that have been waiting for the author to answer questions or for code changes will be labeled as `stale` after a week and will be automatically closed after 2 weeks.
```

Comment:
Presumably this should apply only to PRs with an explicit "waiting-on-author" label? Maybe we should say this? Since obviously a PR could be waiting-on-reviewer for a longer period, and this wouldn't be good but it also is not under the control of the PR author.

Reply:
P.S. specially after reading that the next chapter is entitled **No rush** ...